### PR TITLE
[DP-289] Cleanups + Drop a few Pages / Components

### PR DESCRIPTION
**Pull request by Qubad786**
## [DP-289](https://woven-dojo.atlassian.net/browse/DP-289)

Updates
- Remove `dojo-api`'s catalog API integration
- Drop Programs and ProgramEnrollments
- Drop CourseAssociatedPrograms
- Drop SkillsQuiz
- Drop Pathways
- Drop Program and Pathway Search
- Drop program
- Drop CoursePage (includes Discovery and `edx-enterprise` APIs)
- Drop SearchPage
- Fix tests

### CODE REVIEW

**Comment by bartosz-szczecinski-wp** [comment]
@alain-scialoja-wp you've changed the PR to draft, are we OK to review + merge it? We're past MVP2 so I'd like to merge this ASAP and test if anything broke, else we're going to be stuck in merge conflicts land.

**Comment by Alain1405** [comment]
Sure @bartosz-szczecinski-wp . I think it would be even better to just change the base branch of https://github.com/Woven-dojo/dojo-frontend-app-learner-portal/pull/66 to master and merge all at once.

**Comment by Alain1405** [comment]
> Sure @bartosz-szczecinski-wp . I think it would be even better to just change the base branch of #66 to master and merge all at once.

Also I don't have enough permission to remove the draft.

File: package.json [hunk 1]
```diff
@@ -6,7 +6,6 @@
   "dependencies": {
     "@edx/brand": "npm:@edx/brand-openedx@1.1.0",
     "@edx/frontend-component-footer": "10.1.0",
-    "@edx/frontend-enterprise-catalog-search": "2.7.0",
     "@edx/frontend-enterprise-logistration": "1.1.2",
```
Comment (Alain1405) [inline package.json:9]: I wonder if we need this now that we have our own onboarding experience.

**Comment by bartosz-szczecinski-wp** [comment]
Ok, think we can do what you mentioned (update #66). I'll try to do it as Milan is on planed leave.

File: src/index.jsx [hunk 2]
```diff
@@ -28,7 +28,6 @@ initialize({
     config: () => {
       mergeConfig({
         USE_API_CACHE: process.env.USE_API_CACHE || null,
-        ENTERPRISE_CATALOG_API_BASE_URL: process.env.ENTERPRISE_CATALOG_API_BASE_URL || null,
         ALGOLIA_APP_ID: process.env.ALGOLIA_APP_ID || null,
         ALGOLIA_SEARCH_API_KEY: process.env.ALGOLIA_SEARCH_API_KEY || null,
         ALGOLIA_INDEX_NAME: process.env.ALGOLIA_INDEX_NAME || null,
```
Comment (Alain1405) [inline src/index.jsx:33]: We def don't have algolia dependencies

File: src/smoke.test.jsx [hunk 3]
```diff
@@ -35,7 +35,6 @@ jest.unmock('@edx/frontend-platform/auth');
 
 const mergeTestConfig = () => mergeConfig({
   USE_API_CACHE: process.env.USE_API_CACHE || null,
-  ENTERPRISE_CATALOG_API_BASE_URL: process.env.ENTERPRISE_CATALOG_API_BASE_URL || null,
   ALGOLIA_APP_ID: process.env.ALGOLIA_APP_ID || null,
   ALGOLIA_SEARCH_API_KEY: process.env.ALGOLIA_SEARCH_API_KEY || null,
   ALGOLIA_INDEX_NAME: process.env.ALGOLIA_INDEX_NAME || null,
```
Comment (Alain1405) [inline src/smoke.test.jsx:40]: Should be removed

**Review by Alain1405 — APPROVED** [review]
Approved with some comments of more things that could be removed. Looks GTM otherwise.

File: package.json (hunk 1 above)
Comment (bartosz-szczecinski-wp) [reply package.json:9]: Looks like this is handling some redirection on `/invite/` route so might indeed not be needed. Will note this down and check later.

**Pull request merged**